src/main/java/io/debezium/connector/yugabytedb/YugabyteDBConsistentStreamingSource.java

```diff
@@ -0,0 +1,457 @@
+                        }
+                    }
+
+                    Optional<Message> pollMessage = merger.poll();
+                    while (pollMessage.isPresent()) {
+                        LOGGER.debug("Merger has records");
+                        Message message = pollMessage.get();
+                        CdcService.RowMessage m = message.record.getRowMessage();
+                        YbProtoReplicationMessage ybMessage = new YbProtoReplicationMessage(
+                                m, this.yugabyteDBTypeRegistry);
+                        dispatchMessage(offsetContext, schemaNeeded, recordsInTransactionalBlock,
+                                beginCountForTablet, message.tablet, new YBPartition(message.tableId, message.tablet, false),
+                                message.snapShotTime.longValue(), message.record, m, ybMessage);
+
+                        pollMessage = merger.poll();
+                    }
+
+                    // Reset the retry count, because if flow reached at this point, it means that the connection
+                    // has succeeded
+                    retryCount = 0;
+                }
+            } catch(AssertionError ae) {
+                LOGGER.error("Assertion error received: {}", ae);
+                merger.dumpState();
+
+                // The connector should ideally be stopped if this kind of state is reached.
+                throw new DebeziumException(ae);
+            } catch (Exception e) {
+                ++retryCount;
+                // If the retry limit is exceeded, log an error with a description and throw the exception.
+                if (retryCount > connectorConfig.maxConnectorRetries()) {
+                    LOGGER.error("Too many errors while trying to get the changes from server for tablet: {}. All {} retries failed.", curTabletId, connectorConfig.maxConnectorRetries());
+                    throw e;
+                }
+
+                // If there are retries left, perform them after the specified delay.
+                LOGGER.warn("Error while trying to get the changes from the server; will attempt retry {} of {} after {} milli-seconds. Exception: {}",
+                        retryCount, connectorConfig.maxConnectorRetries(), connectorConfig.connectorRetryDelayMs(), e);
+                LOGGER.warn("Stacktrace", e);
+
+                try {
+                    final Metronome retryMetronome = Metronome.parker(Duration.ofMillis(connectorConfig.connectorRetryDelayMs()), Clock.SYSTEM);
+                    retryMetronome.pause();
+                }
+                catch (InterruptedException ie) {
+                    LOGGER.warn("Connector retry sleep interrupted by exception: {}", ie);
+                    Thread.currentThread().interrupt();
+                }
+            }
+        }
+    }
+
+    private void dispatchMessage(YugabyteDBOffsetContext offsetContext, Map<String, Boolean> schemaNeeded,
```

Comment:
This function should be reusable from Streaming Source as well? Can you add an issue to track the task of reusing this fn?